tempesta_fw/sock_srv.c

```diff
@@ -186,20 +187,24 @@ static int
 tfw_sock_srv_connect_complete(struct sock *sk)
 {
 	int r;
-	TfwSrvConnection *srv_conn = sk->sk_user_data;
-	TfwServer *srv = (TfwServer *)srv_conn->conn.peer;
+	TfwConnection *conn = sk->sk_user_data;
+	TfwServer *srv = (TfwServer *)conn->peer;
+
+	/* Link Tempesta with the socket. */
+	spin_lock(&conn->splock);
+	tfw_connection_link_to_sk(conn, sk);
+	spin_unlock(&conn->splock);
```

Comment:
Again, tfw_connection_link_to_sk() is just a 8-byte assignment. What do you synchronize?